### DISCUSSION TITLE
Mend App for Bitbucket Cloud: Workspace Administration Permissions Bug?

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#32257] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

We currently self-host renovate and have been recently exploring the feasibility of moving to the hosted [Mend App for Bitbucket](https://docs.renovatebot.com/mend-hosted/hosted-apps-config/)

We've installed the app in our Bitbucket Cloud enterprise workspace (of which I am an administrator of via a user-group), as well as my personal workspace (for initial testing)

When going to https://developer.mend.io/bitbucket/{enterprise-workspace} I am not able to see the `Settings` button that I can when comparing to the same app installed for my personal workspace https://developer.mend.io/bitbucket/{personal-workspace}.

This prevents me configuring the workspace wide settings.

Doing some simple debugging, the API call `https://developer.mend.io/api/orgs/bitbucket/{enterprise-workspace}/permissions` returns `{"canAccess":true,"isAdmin":false}`

While checking Bitbucket Cloud directly via `https://api.bitbucket.org/2.0/workspaces/{enterprise-workspace}/permissions?q=permission="owner"` I can confirm that I'm returned as a member of this response.

Interestingly, when going to a specific repository, ie: https://developer.mend.io/bitbucket/{enterprise-workspace}/{some-repo} I am able to see the `Settings` configuration option and `https://developer.mend.io/api/repos/bitbucket/{enterprise-workspace}/{some-repo}/permissions` correctly returns `{"canAccess":true,"isAdmin":true}` as expected.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
We're using this API: https://developer.atlassian.com/cloud/bitbucket/rest/api-group-workspaces/#api-workspaces-workspace-permissions-get

It has three categories, although that docs page even says the middle category is being removed so that leaves them with "owner" and "member". I guess according to that API you would be a member? Then the question is, what is an "admin" in your terminology, or are actually an "owner" via the user group?

**Reply by setchy** [reply]
I can confirm that my user account is part of the API response when calling `https://api.bitbucket.org/2.0/workspaces/{workspace-name}/permissions?q=permission="owner"`, with `permission: "owner"`

**Reply by rarkins** [reply]
Same result when you don't include `?q=permission="owner"`?

**Reply by setchy** [reply]
Yes, same result when calling `https://api.bitbucket.org/2.0/workspaces/{workspace-name}/permissions`, however my user account was on `?page=4` due to the size of our organization directory

```
{
    ...
    "pagelen": 50,
    "size": 775,
    "page": 1,
    ...
 }
 ```
 
Could it be that you're not unpaginating the response?  Passing the `?q-permission="owner"` will save a bunch of API calls while doing so, too

**Reply by setchy** [reply, marked as answer]
After further triage the root was due to workspace level IP allowlisting. 

Docs are being updated to clarify this: https://docs.mend.io/integrations/latest/install-mend-developer-platform-for-bitbucket-clou
